src/medication-dispense/medication-dispense.resource.tsx

```diff
@@ -0,0 +1,85 @@
+import { fhirBaseUrl, openmrsFetch, Session } from "@openmrs/esm-framework";
+import dayjs from "dayjs";
+import useSWR from "swr";
+import { MedicationDispense, MedicationRequest, OrderConfig } from "../types";
+
+export function saveMedicationDispense(
+  medicationDispense: MedicationDispense,
+  abortController: AbortController
+) {
+  return openmrsFetch(`${fhirBaseUrl}/MedicationDispense`, {
+    method: "POST",
+    signal: abortController.signal,
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: medicationDispense,
+  });
+}
+
+export function useOrderConfig() {
+  const { data, error, isValidating } = useSWR<{ data: OrderConfig }, Error>(
+    `/ws/rest/v1/orderentryconfig`,
+    openmrsFetch
+  );
+  return {
+    orderConfigObject: data ? data.data : null,
+    isLoading: !data && !error,
+    isError: error,
+    isValidating,
+  };
+}
+
+export function initiateMedicationDispenseBody(
+  medicationRequests: Array<MedicationRequest>,
+  session: Session
+) {
+  let dispenseBody = [];
+  medicationRequests.map((medicationRequest) => {
+    let dispense = {
+      resourceType: "MedicationDispense",
+      status: "on-hold", // might need to change this to appropriate status
+      medicationReference: medicationRequest.medicationReference,
+      subject: medicationRequest.subject,
+      performer: [
+        {
+          actor: {
+            reference: medicationRequest.requester.reference,
+          },
+        },
+      ],
+      // location: {
+      //   reference: session?.sessionLocation?.uuid,
+      // },
+      type: {
+        coding: [
+          {
+            code: "04affd1a-49ab-44e5-a6d1-c0a3fffceb7d", // what is this?
```

Comment:
This can likely be left blank for now, like the "status", we will likely want to build some logic around this, but I don't think we have figured that out yet.  Where did this uuid come from?

Reply:
The docs has it defined as this 👇🏾  but this looks like something that needs further discussion (definitely not MVP)

> "[type](https://www.hl7.org/fhir/medicationdispense-definitions.html#MedicationDispense.type)" : { [CodeableConcept](https://www.hl7.org/fhir/datatypes.html#CodeableConcept) }, // [Trial fill, partial fill, emergency fill, etc.](http://terminology.hl7.org/3.1.0/ValueSet-v3-ActPharmacySupplyType.html)